[quant][grapphmode][fx][test][refactor] Refactor quantized add test

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43377 [quant][graphmode][fx] Add support for instance_norm
* #43376 [quant][graphmode][fx] Add support for layer_norm
* #43375 [quant][graphmode][fx] Add support for elu
* #43374 [quant][graphmode][fx] Add support for hardswish
* #43373 [quant][graphmode][fx] Add support for mul and mul relu
* **#43372 [quant][grapphmode][fx][test][refactor] Refactor quantized add test**
* #43335 [quant][graphmode][fx] Add support for batchnorm relu
* #43334 [quant][graphmode][fx] Add support for batchnorm
* #43333 [quant][graphmode][fx] Add support for cat
* #43332 [quant][graphmode][fx] Add support for add relu
* #43331 [quant][graphmode][fx] Add support for add
* #43287 [quant][graphmode][fx] Add support for conv module + relu

Summary:
So that adding more binary op tests are easier

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23257046](https://our.internmc.facebook.com/intern/diff/D23257046)

jerryzh168 committed:
[quant][grapphmode][fx][test][refactor] Refactor quantized add test

Summary:
So that adding more binary op tests are easier

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

z-a-f commented on test/quantization/test_quantize_fx.py:
```diff
@@ -209,269 +209,83 @@ def forward(self, x):
 
         #     def forward(self, x, y):
-        #         return x + y
-
-        # class NonQuantizedInplaceAdd(torch.nn.Module):
-        #     def __init__(self):
-        #         super(NonQuantizedInplaceAdd, self).__init__()
-
-        #     def forward(self, x, y):
-        #         x += y
+        #         y = 3 if self.is_scalar else y
+        #         x = self.op(x, y)
         #         return x
```
Do we still need this?

z-a-f commented on test/quantization/test_quantize_fx.py:
```diff
@@ -209,269 +209,83 @@ def forward(self, x):
+        options = itertools.product(
+            [True, False], [True, False], [True, False], [True, False], self.static_quant_types)
+        for is_inplace_op, is_functional_relu, is_inplace_relu, is_scalar, quant_type in options:
+            self.checkGraphModeFxOp(OpRelu(is_inplace_op, is_functional_relu, is_inplace_relu, is_scalar), data, quantized_node, quant_type=quant_type)
```
nit: lint :)

jerryzh168 committed:
Update on "[quant][grapphmode][fx][test][refactor] Refactor quantized add test"


Summary:
So that adding more binary op tests are easier

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23257046](https://our.internmc.facebook.com/intern/diff/D23257046)

[ghstack-poisoned]
